3rdparty/snmalloc/allocator.cpp

```diff
@@ -49,3 +58,14 @@ void oe_allocator_thread_cleanup(void)
     current_alloc_pool()->release(ThreadAlloc::get());
     allocator_local = nullptr;
 }
+
+int oe_allocator_mallinfo(oe_mallinfo_t* info)
+{
+    info->max_total_heap_size = _heap_end - _heap_start;
```

Comment:
Can _heap_end ever decrease under both snmalloc and dlmalloc?

Reply:
That's a good question, and I do not know what the future-proof answer is. Currently though, the allocators are given a fixed start and end via oe_allocator_init(), and there's no API/provision to change it, so the answer is no.

Reply:
Sorry, I intended to ask if **_heap_start** would ever decrease.
My understanding is that both the allocators increment _heap_start to fetch memory from the heap. I don't know if either allocator ever gives back memory to the heap.
Speaking of which, `_heap_end - _heap_start` would give the amount of free memory no?

Reply:
`_heap_start` is not used by `snmalloc` the call to `setup_initial_range` modifies state inside the PAL to account for the range that is not visible to OE.

Reply:
@anakrish I think you're thinking of `_heap_next` for dlmalloc, `_heap_start` is never modified there either. I'll change this to be captured as `total_heap_size` in `oe_allocator_init()` for both allocators, which will hopefully be clearer.

Reply:
Did that on both PRs. The dlmalloc code in this one isn't updated, but will be once it's rebased on the other one of course.

Reply:
Yes,  `_heap_next` was what I was thinking of.